system/git_config.py

```diff
@@ -0,0 +1,207 @@
+    don't work correctly in check mode.
+options:
+  list_all:
+    description:
+      - List all settings (optionally limited to a given scope)
+    required: false
+    default: no
+  name:
+    description:
+      - The name of the setting. If no value is supplied, the value will
+        be read from the config if it has been set.
+    required: false
+    default: null
+  repo:
+    description:
+      - Path to a git repository for reading and writing values from a
+        specific repo.
+    required: false
+    default: null
+  scope:
+    description:
+      - ...
+    required: false
+    default: null
+  value:
+    description:
+      - When specifying the name of a single setting, supply a value to
+        set that setting to the given value.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Set some settings in ~/.gitconfig
+- git_config: name=alias.ci scope=global value=commit
+- git_config: name=alias.st scope=global value=status
+
+# Or system-wide:
+- git_config: name=alias.remotev scope=system value="remote -v"
+- git_config: name=core.editor scope=global value=vim
+# scope=system is the default
+- git_config: name=alias.diffc value="diff --cached"
+- git_config: name=color.ui value=auto
+
+# Make etckeeper not complain when invoked by cron
+- git_config: name=user.email repo=/etc scope=local value="root@{{ ansible_fqdn }}"
+
+# Read individual values from git config
+- git_config: name=alias.ci scope=global
+# scope=system is also assumed when reading values, unless list_all=yes
+- git_config: name=alias.diffc
+
+# Read all values from git config
+- git_config: list_all=yes scope=global
+# When list_all=yes and no scope is specified, you get configuration from all scopes
+- git_config: list_all=yes
+# Specify a repository to include local settings
+- git_config: list_all=yes repo=/path/to/repo.git
+'''
+
+RETURN = '''
+---
+config_value:
+  description: When list_all=no and value is not set, a string containing the value of the setting in name
+  returned: success
+  type: string
+  sample: "vim"
+
+config_values:
+  description: When list_all=yes, a dict containing key/value pairs of multiple configuration settings
+  returned: success
+  type: dictionary
+  sample:
+    core.editor: "vim"
+    color.ui: "auto"
+    alias.diffc: "diff --cached"
+    alias.remotev: "remote -v"
+'''
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            list_all=dict(required=False, type='bool', default=False),
+            name=dict(type='str'),
+            repo=dict(type='str'),
+            scope=dict(required=False, type='str', choices=['local', 'global', 'system']),
+            value=dict(required=False)
+        ),
+        mutually_exclusive=[['list_all', 'name'], ['list_all', 'value']],
+        required_if=[('scope', 'local', ['repo'])],
+        required_one_of=[['list_all', 'name']],
+        supports_check_mode=True,
+    )
+    git_path = module.get_bin_path('git')
```

Comment:
Need to deal with the case if git isn't found, currently this fails with

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: sequence item 0: expected string, NoneType found
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "parsed": false}
```

Have at https://github.com/ansible/ansible-modules-extras/blob/devel/system/puppet.py#L160 for how to gracefully fail